.pre-commit-config.yaml

```
@@ -40,7 +40,30 @@ repos:
     hooks:
       - id: regenerate-files
         name: regenerate generated files
-        language: system
+        language: python
         entry: python src/trio/_tools/gen_exports.py
         pass_filenames: false
+        additional_dependencies: ["astor", "attrs", "black", "ruff"]
         files: ^src\/trio\/_core\/(_run|(_i(o_(common|epoll|kqueue|windows)|nstrumentation)))\.py$
+  - repo: https://github.com/astral-sh/uv-pre-commit
+    rev: 0.5.9
+    hooks:
+      # Compile requirements
+      - id: pip-compile
+        name: uv pip-compile test-requirements.in
+        args: [
+          "--universal",
+          "--python-version=3.9",
+          "test-requirements.in",
+          "-o",
+          "test-requirements.txt"]
```

I think this reads better:
```suggestion
        args:
        - --universal
        - --python-version=3.9
        - --output-file=test-requirements.txt
        - test-requirements.in
```

---

That does look a bit nicer, I agree

---

in a python repo I'd argue that using python-looking syntax is going to be easier for people to understand.

---

That is a pretty good argument actually

---

Dunno, it's usually harder for me as it looks like type annotations, but it isn't.

---

I think we should use a yaml formatter

---

see #3164, I'll do whatever for this PR and then it can be enforced later.